include/nuttx/net/usrsock.h

```diff
@@ -83,6 +83,7 @@ enum usrsock_request_types_e
   USRSOCK_REQUEST_ACCEPT,
   USRSOCK_REQUEST_IOCTL,
   USRSOCK_REQUEST_SHUTDOWN,
+  USRSOCK_REQUEST_AVAILABLE,
```

Comment:
why not reuse USRSOCK_REQUEST_SOCKET? usersock server could return a special error code to continue the normal creation process.

Reply:
@xiaoxiang781216 san
In my understand, USRSOCK_REQUEST_SOCKET is just for setup usrsock daemon socket.
I want to tell usrsock can use socket for internet connection or not. So, I added a new I/F.

Reply:
please add this information to the commit message

Reply:
> please add this information to the commit message

I see, thank you!

Reply:
I see, I will update commit message.

Reply:
@jerpelea san,
I updated a commit message.

Reply:
> @xiaoxiang781216 san In my understand, USRSOCK_REQUEST_SOCKET is just for setup usrsock daemon socket, and it is done by socket.c. I want to tell usrsock can use socket for internet connection or not. So, I added a new I/F.

it is no much difference here. We can move the call of si_setup from psock_socket to net_sockif and:

1. Try to call g_usrsock_sockif->si_setup first
2. continue the normal path if fail

maybe we can change from:
```
FAR const struct sock_intf_s *
net_sockif(sa_family_t family, int type, int protocol);
```
to:
```
int net_setup(FAR struct socket *psock);
```

Reply:
@xiaoxiang781216 san
Sorry for late reply.
Yes, it is feasible on both sides of that as you have pointed out.
However, I think it would be difficult to understand if we use si_setup() errors to fallback, such that a part of error are returned as is depending on the error number, and the Kernel stack is used for errors when the user specifies that the error is prohibited.
Therefore, I added a new interface.

Reply:
this can be handled by checking special error code, e.g. -ENOSYS or -ENOSUP.

Reply:
BTW, using si_setup could make other usrsock implementation support the same function more quickly.

Reply:
@xiaoxiang781216 san
Thank you for your recommendation.
I'll change it.